tests/integration/workflows/python_pip/test_python_pip.py

```diff
@@ -137,7 +138,15 @@ def test_mismatch_runtime_python_project(self):
                 runtime=self.runtime_mismatch[self.runtime],
             )
         except WorkflowFailedError as ex:
-            self.assertIn("Binary validation failed", str(ex))
+            # handle both e.g. missing /usr/bin/python2.7 and situation where
+            # python2.7 does not have pip installed (as is the case in some
+            # Mac environments)
+            ex_s = str(ex)
+            if (
+                "Binary validation failed" not in ex_s
+                and "pip executable not found in your python environment" not in ex_s
+            ):
+                six.raise_from(AssertionError("Unexpected exception"), ex)
```

Comment:
nit: Consider not using `six`

 ```suggestion
                raise AssertionError("Unexpected exception") from ex
```